test/controllers/api/v2/usergroups_controller_test.rb

```diff
@@ -29,15 +29,108 @@ def setup
+  end
+
+  test "should create usergroup with usergroup" do
+    post :create, params: { :usergroup => valid_attrs.clone.update(:usergroup_ids => [@usergroup.id])}
+    assert_response :created
+    assert_equal 1, JSON.parse(@response.body)["usergroups"].length
+    assert_equal JSON.parse(@response.body)["usergroups"][0]["name"], @usergroup.name, "Can't create usergroup with usergroup #{@usergroup}"
+  end
+
   test "should update usergroup" do
     put :update, params: { :id => @usergroup.to_param, :usergroup => valid_attrs }
     assert_response :success
   end
 
+  test "should update usergroup with role" do
+    role = Role.find_by_name('Manager')
+    put :update, params: { :id => @usergroup.to_param, :usergroup => valid_attrs.clone.update(:role_ids => [role.id]) }
+    assert_response :success
+    assert_equal role.name, JSON.parse(@response.body)["roles"][0]["name"], "Can't update usergroup with role #{role}"
+  end
+
+  test "should update usergroup with user" do
+    user = User.find_by_login('one')
+    put :update, params: { :id => @usergroup.to_param, :usergroup => valid_attrs.clone.update(:user_ids => [user.id]) }
+    assert_response :success
+    assert_equal JSON.parse(@response.body)["users"][0]["login"], user.login, "Can't update usergroup with user #{user}"
+  end
+
+  test "should update usergroup with usergroup" do
+    usergroup  = FactoryBot.create(:usergroup)
+    put :update, params: { :id => @usergroup.to_param, :usergroup => valid_attrs.clone.update(:usergroup_ids => [usergroup.id]) }
+    assert_response :success
+    assert_equal JSON.parse(@response.body)["usergroups"][0]["name"], usergroup.name, "Can't update usergroup with user #{usergroup}"
+  end
+
   test "should destroy usergroups" do
     assert_difference('Usergroup.count', -1) do
       delete :destroy, params: { :id => @usergroup.to_param }
     end
     assert_response :success
   end
+
+  test "should not create usergroup with invalid name" do
+    post :create, params: { :usergroup => { :name => '' } }
+    assert_response :unprocessable_entity, "Can create usergroup with empty name"
+  end
+
+  test "should not create usergroup with same name" do
+    post :create, params: { :usergroup => { :name => @usergroup.name } }
+    assert_response :unprocessable_entity, "Can create usergroup with already taken name"
+  end
+
+  test "should not update usergroup with invalid name" do
+    put :update, params: { :id => @usergroup.id, :usergroup => { :name => '' } }
+    assert_response :unprocessable_entity, "Can update usergroup with empty name"
+    @usergroup.reload
+    assert_not_equal @usergroup.name, ''
+  end
+
+  test "should not update usergroup with same name" do
+    usergroup = FactoryBot.create(:usergroup)
+    put :update, params: { :id => usergroup.id, :usergroup => { :name => @usergroup.name } }
+    assert_response :unprocessable_entity, "Can update usergroup with already taken name"
```

Comment:
Shouldn't all these be really "Can't"?

Reply:
Logically it's correct because if we can update usergroup with empty or already taken name, error message will tell the same :)
Error message in test should be negative test name:
should create -> can't create
should update -> can't update
should not update -> can update
should not create -> can create

Reply:
I see :+1: